Sdks 3728 swift 6 support

vahancouver opened the pull request:
[SDKS-3728](https://pingidentity.atlassian.net/browse/SDKS-3728) [iOS] Revisit Swift 6 support (part 1/2)

Update the following modules to support Swift 6 and complete strict concurrency mode
- Logger
- Storage
- Orchestrate

george-bafaloukas-forgerock commented on Orchestrate/Orchestrate/CookieModule.swift:
```diff
@@ -175,19 +182,19 @@ extension Workflow {
 
 /// A storage class for managing in-memory cookies.
-public final class InMemoryCookieStorage: HTTPCookieStorage {
+public actor InMemoryCookieStorage {
```
If we intent to keep this as final I suggest we add `final` here as well. According to Swift forums while actors do not implement inheritance, this might change in the future. Xcode will probably not have an issue with that.
https://forums.swift.org/t/why-can-you-constrain-to-final-classes-and-actors/65256/3

vahancouver replied:
Added the `final` just in case

george-bafaloukas-forgerock commented on Orchestrate/Orchestrate/HttpClient.swift:
```diff
@@ -13,10 +13,10 @@ import Foundation
 /// `HttpClient` is responsible for handling HTTP requests and logging the details of those requests and responses.
-public class HttpClient {
-    var session: URLSession
+public final class HttpClient: Sendable {
```
Do we need to restrict this to final? Could we benefit from making this an open class for further customization?

vahancouver replied:
yeah that sounds good. made it open with `@unchecked Sendable`

george-bafaloukas-forgerock reviewed (commented):
Changes look good to me. We need to figure out the tests on the CI to make sure nothing else is broken though. Few questions/comment left for discussion

vahancouver commented:
> Changes look good to me. We need to figure out the tests on the CI to make sure nothing else is broken though. Few questions/comment left for discussion

@george-bafaloukas-forgerock Addressed the comments and updated the github workflow to use newer xcode and simulator that supports Swift 6. All tests are passing both locally and on github

spetrov reviewed (approved):
Looks good to me!

rodrigoareis reviewed (approved):
Changes looks good to me
